lib/srv/db/cloud/watchers/rdsproxy.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watchers
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/aws/aws-sdk-go/service/rds/rdsiface"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+// rdsDBProxyFetcher retrieves RDS proxies and proxy endpoints.
+type rdsDBProxyFetcher struct {
+	cfg rdsFetcherConfig
+	log logrus.FieldLogger
+}
+
+// newRDSDBProxyFetcher returns a new RDS proxy fetcher instance.
+func newRDSDBProxyFetcher(config rdsFetcherConfig) (Fetcher, error) {
+	if err := config.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &rdsDBProxyFetcher{
+		cfg: config,
+		log: logrus.WithFields(logrus.Fields{
+			trace.Component: "watch:rdsproxy",
+			"labels":        config.Labels,
+			"region":        config.Region,
+		}),
+	}, nil
+}
+
+// Get returns RDS proxies and prosy endpoints matching the watcher's
+// selectors.
+func (f *rdsDBProxyFetcher) Get(ctx context.Context) (types.Databases, error) {
+	databases, err := f.getRDSProxyDatabases(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var result types.Databases
+	for _, database := range databases {
+		match, _, err := services.MatchLabels(f.cfg.Labels, database.GetAllLabels())
+		if err != nil {
+			f.log.Warnf("Failed to match %v against selector: %v.", database, err)
+		} else if match {
+			result = append(result, database)
+		} else {
+			f.log.Debugf("%v doesn't match selector.", database)
+		}
+	}
+	return result, nil
+}
+
+// getRDSProxyDatabases returns a list of database resources representing RDS
+// proxies and proxy endpoints.
+func (f *rdsDBProxyFetcher) getRDSProxyDatabases(ctx context.Context) (types.Databases, error) {
+	rdsProxies, err := getRDSProxies(ctx, f.cfg.RDS, maxPages)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	rdsProxyEndpoints, err := getRDSProxyEndpoints(ctx, f.cfg.RDS, maxPages)
+	if err != nil {
+		f.log.Debugf("Failed to get RDS proxy endpoints: %v.", err)
```

Comment:
Is this failure considered non-critical on purpose? Explain with a comment if so.